resolve_test.go

```diff
@@ -0,0 +1,283 @@
+package gomatrixserverlib
+
+import (
+	"context"
+	"fmt"
+	"net"
+	"reflect"
+	"testing"
+
+	"github.com/miekg/dns"
+	"gopkg.in/h2non/gock.v1"
+)
+
+const (
```

Comment:
it might be good to have some tests for ipv6 things here too.

Reply:
Why? Apart from the SRV lookup there's no DNS lookup involved on gomatrixserverlib's side because that being taken care of by the core HTTP library.

Reply:
well, sure, but is a server name of `[1::1:1]` correctly handled as an IPv6 literal with no port? Are all three result fields set correctly? 

How about `[ab::cd]:12` ?

Reply:
Aaah I didn't understand you correctly. Sure, I'll add some.